File: CF Python/M.py
kx, ky = map(int, input().split())
n = int(input())

count_x = {x: 0 for x in range(1, kx + 1)}
count_y = {y: 0 for y in range(1, ky + 1)}
count_xy = {}

for i in range(n):
    x, y = map(int, input().split())
    count_x[x] += 1
    count_y[y] += 1
    if (x, y) in count_xy:
        count_xy[(x, y)] += 1
    else:
        count_xy[(x, y)] = 1

x0 = {x: n for x in range(1, kx + 1)}
s = 0
for (x, y), count in count_xy.items():
    p = count_x[x] * count_y[y] / n
    s += (count - p) ** 2 / p
    x0[x] -= count_y[y]

for x in range(1, kx + 1):
    s += x0[x] / n * count_x[x]

print(s)

File: CF PABC.NET/M.pas
begin
  var (KX, KY) := ReadlnInteger2();
  var N := ReadlnInteger();
  
  var CountX := Range(1, KX).ToDictionary(K -> K, V -> 0);
  var CountY := Range(1, KY).ToDictionary(K -> K, V -> 0);
  var CountXY := new Dictionary<KeyValuePair<Integer, Integer>, Integer>;
  
  for var I := 1 to N do
  begin
    var (X, Y) := ReadlnInteger2();
    CountX[X] += 1;
    CountY[Y] += 1;
    var XY := KV(X, Y);
    if CountXY.Keys.Contains(XY) then
      CountXY[XY] += 1
    else
      CountXY.Add(XY, 1);
  end;
  
  var X0 := Range(1, KX).ToDictionary(K -> K, V -> N);
  var S := 0.0;
  foreach var XY in CountXY do
  begin
    var Count := XY.Value;
    var (X, Y) := (XY.Key.Key, XY.Key.Value);
    var P := CountX[X] / N * CountY[Y];
    S += (Count - P).Sqr / P;
    X0[X] -= CountY[Y];
  end;
  
  for var X := 1 to KX do
    S += X0[X] / N * CountX[X];
  
  Writeln(S:0:9);
end.